libraries/src/Document/HtmlDocument.php

```diff
@@ -778,6 +787,50 @@ protected function _fetchTemplate($params = [])
         return $this;
     }
 
+    /**
+     * Returns a toolbar object or null
+     *
+     * @param $toolbar
+     * @return Toolbar|null
+     */
+    public function getToolbar($toolbar = 'toolbar', $create = true): ?Toolbar
+    {
+        if (empty($this->toolbars[$toolbar])) {
+            if (!$create) {
+                return null;
+            }
+
+            $this->toolbars[$toolbar] = CmsFactory::getContainer()->get(ToolbarFactoryInterface::class)->createToolbar($toolbar);
+        }
+
+        return $this->toolbars[$toolbar];
+    }
+
+    /**
+     * Returns the toolbar array
+     *
+     * @return array
+     */
+    public function getToolbars(): array
+    {
+        return $this->toolbars;
+    }
+
+    /**
+     * Adds a new or replace an existing toolbar object
+     *
+     * @param $name
+     * @param Toolbar $toolbar
+     *
+     * @return $this
+     */
+    public function setToolbar($name, Toolbar $toolbar): self
```

Comment:
```suggestion
     * @param   string   $name
     * @param   Toolbar  $toolbar
     *
     * @return  $this
     *
     * @since   __DEPLOY_VERSION__
     */
    public function setToolbar(string $name, Toolbar $toolbar): self
```
since tag

should `$name` forced to be a string?